### ISSUE TITLE
docs(gp): warn that DistLikelihood silently freezes parameterized link functions

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/pyrox#202]
## Summary

`DistLikelihood.dist_fn` is declared `eqx.field(static=True)` (`_likelihoods.py:90`). Anything the callable closes over is therefore invisible to `eqx.filter_grad`, so a link function carrying learnable parameters **never trains** — with no error, no warning, and a loss that still decreases because the kernel hyperparameters and guide keep fitting.

Add a warning to the docstring and a test that pins the behaviour.

This is independent of any other work and worth landing on its own.

## Reproduction

```python
import equinox as eqx, jax, jax.numpy as jnp

class Warp(eqx.Module):
    a: jnp.ndarray
    def __call__(self, f):
        return jnp.sinh(self.a * jnp.arcsinh(f))

warp = Warp(a=jnp.asarray(1.3))
lik = DistLikelihood(lambda f: dist.Normal(warp(f), 0.5))

grad = eqx.filter_grad(lambda L: -L.log_prob(f, y))(lik)
leaves = jax.tree_util.tree_leaves(eqx.filter(grad, eqx.is_inexact_array))
# len(leaves) == 0  -> warp.a is frozen forever
```

*Confirmed:* `eqx.filter_grad` returns **zero trainable leaves**. The parameters are not merely un-updated; they are absent from the gradient pytree entirely.

## Why this matters beyond one class

The trap applies to *any* parameterized link, not just warps: a learnable dispersion in a Negative-Binomial link, a learnable degrees-of-freedom in a Student-t link, a calibration temperature. Every one of these is a natural thing to reach for and every one fails the same silent way.

`DistLikelihood`'s existing examples all use fixed link functions
(`lambda f: dist.Bernoulli(logits=f)`, `lambda f: dist.Poisson(rate=jnp.exp(f))`), which is why nothing has surfaced this.

## Fix

Add to the `DistLikelihood` class docstring, after the existing examples:

```
    !!! warning "The link function cannot carry trainable parameters"
        ``dist_fn`` is a **static** field, so anything the callable closes
        over is invisible to ``eqx.filter_grad``. A link function holding
        learnable parameters will silently never train — there is no error,
        and the loss still decreases because the kernel and guide keep
        fitting.

        ```python
        # WRONG -- `warp` is frozen at its initial value forever
        lik = DistLikelihood(lambda f: dist.Normal(warp(f), sigma))
        ```

        For a parameterized link, write a `Likelihood` subclass and
        hold the parameters as child modules, so they are ordinary
        trainable leaves.
```

Keep the guidance concrete — "write a `Likelihood` subclass" is the actual remedy and takes about ten lines.

## Acceptance criteria

Add to `packages/pyrox-gp/tests/gp/test_likelihoods.py`.

- [ ] **Pin the freezing behaviour.** Build a `DistLikelihood` whose `dist_fn`
      closes over an `eqx.Module` with an inexact-array field. Assert
      `jax.tree_util.tree_leaves(eqx.filter(grad, eqx.is_inexact_array))` is
      **empty**. The test documents the trap rather than asserting it is fixed —
      name it accordingly (e.g. `test_dist_fn_is_static_and_freezes_closures`)
      and give it a docstring pointing at the class warning.
- [ ] **Contrast case.** A minimal `Likelihood` subclass holding the same
      parameter as a child module produces a **non-empty**, finite gradient.
      This is what makes the test informative rather than just a snapshot of
      current behaviour.
- [ ] Docstring warning renders in mkdocstrings — `!!! warning` admonition,
      Google-style, **no Sphinx/RST markup** (`test_docstrings.py` rejects it).

## Explicitly not doing

- **Do not change `dist_fn` to a non-static field.** It is static for a reason:
  a Python callable is not a valid JAX pytree leaf, and making it dynamic would
  break `jit` on every existing use. The static declaration is correct; only its
  consequence is undocumented.
- Do not add runtime detection of closed-over arrays. It would be fragile
  (inspecting closures), and the subclass remedy is simple enough that a
  docstring is the right level of intervention.

## Before reporting done

Per `AGENTS.md`:

```bash
uv run pytest -q
uv run --group lint ruff check .          # note: '.', not 'packages/'
uv run --group lint ruff format --check .
uv run --group typecheck ty check packages/pyrox-gp/src/pyrox_gp
```
